Compile failures with `-fno-lax-vector-conversions`

Command: `clang simde/arm/neon/qsub.h  -Werror -O0 -Wno-c++11-narrowing -I.`

Seems like there is an issue here:
https://github.com/simd-everywhere/simde/blob/master/simde/arm/neon/qsub.h#L331

```
simde/arm/neon/qsub.h:331:17: error: cannot convert between vector type '__attribute__((__vector_size__(2 * sizeof(int)))) int' (vector of 2 'int' values) and vector type '__attribute__((__vector_size__(2 * sizeof(uint32_t)))) uint32_t' (vector of 2 'uint32_t' values) as implicit conversion would cause truncation
      r_.values &= (r_.values <= a_.values);
                ^
simde/arm/neon/qsub.h:360:17: error: cannot convert between vector type '__attribute__((__vector_size__(1 * sizeof(long long)))) long long' (vector of 1 'long long' value) and vector type '__attribute__((__vector_size__(1 * sizeof(uint64_t)))) uint64_t' (vector of 1 'uint64_t' value) as implicit conversion would cause truncation
      r_.values &= (r_.values <= a_.values);
                ^
simde/arm/neon/qsub.h:666:17: error: cannot convert between vector type '__attribute__((__vector_size__(2 * sizeof(long long)))) long long' (vector of 2 'long long' values) and vector type '__attribute__((__vector_size__(2 * sizeof(uint64_t)))) uint64_t' (vector of 2 'uint64_t' values) as implicit conversion would cause truncation
      r_.values &= (r_.values <= a_.values);
```

Do we need a reinterpret cast here?

--- a/simde/arm/neon/cvt.h
+++ b/simde/arm/neon/cvt.h
@@ -630,7 +630,7 @@ simde_vcvtq_u64_f64(simde_float64x2_t a) {
       static const double min_representable SIMDE_VECTOR(16) = { SIMDE_FLOAT64_C(0.0), };
       r_.values &= HEDLEY_REINTERPRET_CAST(__typeof__(r_.values), a_.values > min_representable);
 
-      r_.values &= (a_.values == a_.values);
+      r_.values &= HEDLEY_REINTERPRET_CAST(__typeof__(r_.values), (a_.values == a_.values));
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.values) / sizeof(r_.values[0])) ; i++) {

--- a/simde/arm/neon/qadd.h
+++ b/simde/arm/neon/qadd.h
@@ -715,7 +715,7 @@ simde_vqaddq_u32(simde_uint32x4_t a, simde_uint32x4_t b) {
       r_.m128i = _mm_or_si128(_mm_cmpgt_epi32(a_.m128i, _mm_xor_si128(i32min, sum)), sum);
     #elif defined(SIMDE_VECTOR_SUBSCRIPT)
       r_.values = a_.values + b_.values;
-      r_.values |= HEDLEY_STATIC_CAST(__typeof__(r_.values), r_.values < a_.values);
+      r_.values |= HEDLEY_REINTERPRET_CAST(__typeof__(r_.values), r_.values < a_.values);
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.values) / sizeof(r_.values[0])) ; i++) {

--- a/simde/arm/neon/qsub.h
+++ b/simde/arm/neon/qsub.h
@@ -268,7 +268,7 @@ simde_vqsub_u8(simde_uint8x8_t a, simde_uint8x8_t b) {
       r_.m64 = _mm_subs_pu8(a_.m64, b_.m64);
     #elif defined(SIMDE_VECTOR_SUBSCRIPT_SCALAR)
       r_.values  = a_.values - b_.values;
-      r_.values &= (r_.values <= a_.values);
+      r_.values &= HEDLEY_REINTERPRET_CAST(__typeof__(r_.values), (r_.values <= a_.values));
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.values) / sizeof(r_.values[0])) ; i++) {
@@ -299,7 +299,7 @@ simde_vqsub_u16(simde_uint16x4_t a, simde_uint16x4_t b) {
       r_.m64 = _mm_subs_pu16(a_.m64, b_.m64);
     #elif defined(SIMDE_VECTOR_SUBSCRIPT_SCALAR)
       r_.values  = a_.values - b_.values;
-      r_.values &= (r_.values <= a_.values);
+      r_.values &= HEDLEY_REINTERPRET_CAST(__typeof__(r_.values), (r_.values <= a_.values));
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.values) / sizeof(r_.values[0])) ; i++) {
@@ -328,7 +328,7 @@ simde_vqsub_u32(simde_uint32x2_t a, simde_uint32x2_t b) {
 
     #if defined(SIMDE_VECTOR_SUBSCRIPT_SCALAR)
       r_.values  = a_.values - b_.values;
-      r_.values &= (r_.values <= a_.values);
+      r_.values &= HEDLEY_REINTERPRET_CAST(__typeof__(r_.values), (r_.values <= a_.values));
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.values) / sizeof(r_.values[0])) ; i++) {
@@ -357,7 +357,7 @@ simde_vqsub_u64(simde_uint64x1_t a, simde_uint64x1_t b) {
 
     #if defined(SIMDE_VECTOR_SUBSCRIPT_SCALAR)
       r_.values  = a_.values - b_.values;
-      r_.values &= (r_.values <= a_.values);
+      r_.values &= HEDLEY_REINTERPRET_CAST(__typeof__(r_.values), (r_.values <= a_.values));
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.values) / sizeof(r_.values[0])) ; i++) {
@@ -631,10 +631,10 @@ simde_vqsubq_u32(simde_uint32x4_t a, simde_uint32x4_t b) {
         );
     #elif defined(SIMDE_VECTOR_SUBSCRIPT_SCALAR)
       r_.values  = a_.values - b_.values;
-      r_.values  = r_.values & (r_.values <= a_.values);
+      r_.values &= HEDLEY_REINTERPRET_CAST(__typeof__(r_.values), (r_.values <= a_.values));
     #elif defined(SIMDE_VECTOR_SUBSCRIPT_SCALAR)
       r_.values  = a_.values - b_.values;
-      r_.values &= (r_.values <= a_.values);
+      r_.values &= HEDLEY_REINTERPRET_CAST(__typeof__(r_.values), (r_.values <= a_.values));
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.values) / sizeof(r_.values[0])) ; i++) {
@@ -663,7 +663,7 @@ simde_vqsubq_u64(simde_uint64x2_t a, simde_uint64x2_t b) {
 
     #if defined(SIMDE_VECTOR_SUBSCRIPT_SCALAR)
       r_.values  = a_.values - b_.values;
-      r_.values &= (r_.values <= a_.values);
+      r_.values &= HEDLEY_REINTERPRET_CAST(__typeof__(r_.values), (r_.values <= a_.values));
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.values) / sizeof(r_.values[0])) ; i++) {

